server/src/main/java/org/opensearch/index/engine/exec/DataFormat.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.engine.exec;
+
+import org.opensearch.common.settings.Setting;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.index.engine.exec.text.TextDF;
+
+public interface DataFormat {
```

Comment:
How does a DataFormat indicate the field type/s that it is supporting ? 

Like Parquet indicating that it stores only Numeric types, Lucene indicates it stores text; etc.